pkg/stub/handler.go

```diff
@@ -1,102 +1,119 @@
 package stub
 
 import (
-	"context"
+	"time"
 
 	"github.com/aerogear/mobile-developer-console/pkg/apis/aerogear/v1alpha1"
 	"github.com/aerogear/mobile-developer-console/pkg/mobile"
-	"github.com/operator-framework/operator-sdk/pkg/sdk"
-	log "github.com/sirupsen/logrus"
+	"github.com/labstack/gommon/log"
 	"k8s.io/api/core/v1"
 )
 
-func NewHandler(mobileAppRepo mobile.MobileClientRepo) sdk.Handler {
-	return &Handler{
-		mobileClientRepo: mobileAppRepo,
+func WatchSecrets(namespace string, secretsCRUDL mobile.SecretsCRUDL, mobileAppRepo mobile.MobileClientRepo) {
```

Comment:
I think without too much effort, we can refactor this function a bit to improve the reusability: it can take in a `watchInterface`, and a listener function. Then this function can invoke the passed in `watchInteferace` to start watching, and invoke the listener function when it should be called.

Reply:
makes sense, will refactor it